cookbook/examples/agents/thinking_agent_custom_tool_decorator.py

```diff
@@ -0,0 +1,39 @@
+import httpx
+from agno.agent import Agent
+from agno.tools import tool
+
+
+@tool(
+    name="fetch_hackernews_stories",
+    description="Get top stories from Hacker News",
+    show_result=True,
+    instructions="When using the Hacker News API:\n1. Present stories in a clean, readable format\n2. Provide context when a URL is missing\n3. Be aware of rate limiting - results are cached for 1 hour",
```

Comment:
The instructions should rather be about when to use this tool. The AI doesn't have control about what goes on inside the tool. Think of it like an extend docstring. 

On the toolkit level it makes more sense because there you give instructions over all the tools in the toolkit. Here it only applies to a single tool.